code/API_definitions/number_verification.yaml

```diff
@@ -0,0 +1,375 @@
+openapi: 3.0.3
+info:
+  title: Number Verification
+  description: |-
+    Service Enabling Network Function API to verify that the provided **phone number** is the one used in the device. It verifies that the user is using a device with the same *phone number* as it is declared.
+    It also make it possible for a Service provider to verify the number itself by returning the phone number retrieved from the authenticated user's access token.
+
+    # Relevant  Definitions and concepts
+    - **NaaS**: *Network-as-a-Service* model where Telco Network resources are exposed to third parties through APIs. In this particular API, Number Verification is exposed following this model. 
+
+    # API Functionality 
+    It enables a Service Provider (SP) to verify the phone number (MSISDN) of the mobile device being used to access their service where the mobile device is accessing the *service provider* over a mobile network (WiFi connections are out of this API scope) either by getting the comparison result or receiving the phone number of the device that it is used, so they can verify it themselves.
+
+    # Resources and Operations overview
+    This API currently provides two endpoints where both require a **3-legged token** and authentication via **mobile network** (excluding for example by SMS/OTP or user/password as an authentication method):
+    - The first one checks if the user phone number matches the MSISDN associated with the mobile device. It can receive either a hashed phone number or a clear phone number as input and it compares the received input with the authenticated user's MSISDN retrieved from the access token in order to respond **true/false**.
+    - The next one retrieves the phone number associated to the user's token and returns it so the verification can be made by the service provider.
```

Comment:
I'd stick to one naming convention, phone number does not equal MSISDN, therefore it can be misleading to audience, e.g. phone number can be fixed number and mobile number. I recommend to use "msisdn" wherever you call phone number or provide a definition of phone number

Reply:
I added this to the API description to be more clear that we're refering to the cell phone number not a landline:
_In this API **phone number** term refers to the mobile phone number (MSISDN)_

Reply:
thx!